client/tx/tx.go

```diff
@@ -265,7 +267,17 @@ func BuildSimTx(txf Factory, msgs ...sdk.Msg) ([]byte, error) {
 		return nil, err
 	}
 
-	simReq := sim.SimulateRequest{Tx: txb.GetProtoTx()}
+	any, ok := txb.(codectypes.IntoAny)
+	if !ok {
+		return nil, fmt.Errorf("cannot simulate tx that cannot be wrapped into any")
+	}
+	cached := any.AsAny().GetCachedValue()
+	txTx, ok := cached.(*tx.Tx)
```

Comment:
This name is not ideal. `tx.Tx` is an interface, correct? Can we call this variable `txI`?

Reply:
No, `sdk.Tx` is the interface. `tx.Tx` is the proto.Message. I renamed it to `protoTx`.